### PR TITLE
Added goal and ball particles. Created custom scene to demo them and ball prefab.

### PR DESCRIPTION
**Pull request by doughtnerd**
Added some of the more refined particles and added them in a scene called Chris C Scene.
Created a prefab for the Particle Ball in the Interactables folder.
Created TriggerCollisionParticle script to trigger particles like sparks or the goals on collision/trigger. Script is inside Chris C Particles/Scripts folder.

### CODE REVIEW

**Comment by ChristianIsaacRoy** [comment]
TriggerCollisionParticle.cs review
using System.Collections.Generic is not needed and can be removed.

Add a region for Monobehaviour methods OnCollision, OnTrigger, and OnDisable
Add a region for private helpers CheckandPlay, playParticleSystem
Add explicit "private" keyword before PlayParticleSystem

File: Ricochet/Assets/_Art/Chris C Particle Effects/Scripts/TriggerCollisionParticle.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+﻿using System.Collections;
+using System.Collections.Generic;
```
Comment (ChristianIsaacRoy) [inline Ricochet/Assets/_Art/Chris C Particle Effects/Scripts/TriggerCollisionParticle.cs]: using system.collections.generic can be deleted. Not needed

File: Ricochet/Assets/_Art/Chris C Particle Effects/Scripts/TriggerCollisionParticle.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+using UnityEngine;
+
+namespace CCParticles
+{
+    public class TriggerCollisionParticle : MonoBehaviour
+    {
+        #region Inspector Variables
+
+        [SerializeField]
+        private ParticleSystem particleSystem;
```
Comment (ChristianIsaacRoy) [inline Ricochet/Assets/_Art/Chris C Particle Effects/Scripts/TriggerCollisionParticle.cs]: overriding a variable in monobehaviour, did you want to change the name of the variable or use the "new" keyword to show that overriding the variable was intended.

File: Ricochet/Assets/_Art/Chris C Particle Effects/Scripts/TriggerCollisionParticle.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            this.particleSystem.gameObject.SetActive(false);
+        }
+
+        private void CheckAndPlay(Collider2D collision)
+        {
+            if (this.particleSystem != null)
+            {
+                if (((1 << collision.gameObject.layer) & this.collisionLayer) != 0)
+                {
+                    StartCoroutine(PlayParticleSystem(this.particleSystem));
```
Comment (ChristianIsaacRoy) [inline Ricochet/Assets/_Art/Chris C Particle Effects/Scripts/TriggerCollisionParticle.cs]: Most of the time, when the ball is scored and collides with the goal, the ball is deactivated before this get's called and an error is thrown.

File: Ricochet/Assets/_Art/Chris C Particle Effects/Scripts/TriggerCollisionParticle.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        #region Inspector Variables
+
+        [SerializeField]
+        private ParticleSystem particleSystem;
+
+        [SerializeField]
+        private LayerMask collisionLayer;
+
+        #endregion
+
```
Comment (ChristianIsaacRoy) [inline Ricochet/Assets/_Art/Chris C Particle Effects/Scripts/TriggerCollisionParticle.cs]: OnCollision, OnTrigger, and OnDisable should be in #region monobehaviour

File: Ricochet/Assets/_Art/Chris C Particle Effects/Scripts/TriggerCollisionParticle.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        private void OnTriggerEnter2D(Collider2D collision)
+        {
+            CheckAndPlay(collision);
+        }
+
+        private void OnDisable()
+        {
+            this.particleSystem.gameObject.SetActive(false);
+        }
+
```
Comment (ChristianIsaacRoy) [inline Ricochet/Assets/_Art/Chris C Particle Effects/Scripts/TriggerCollisionParticle.cs]: checkandplay and playparticlesystem should be in a #region Private Helpers

**Comment by doughtnerd** [comment]
Changes Have been made

**Pull request merged**